tests/settings.py

```diff
@@ -117,7 +117,7 @@
         'rest_framework.authentication.SessionAuthentication',
     ),
     'DEFAULT_FILTER_BACKENDS': (
-        'rest_framework.filters.DjangoObjectPermissionsFilter',
+        'resolwe.permissions.filters.ResolwePermissionsFilter',
```

Comment:
I have this crazy idea. Why don't we leave the original `DjangoObjectPermissionsFilter`. This would work as intended for all our models except Storage, right. For Storage it would return all objects. Then, just for Storage, we add another StoragePermissionsFilter after all the other filters. This filter can be specific for Storage but much shorter then `get_objects_for_user `, and it can filter-out the objects missing permissions.